### PR TITLE
docs: add modules introduction

### LINKED ISSUE #28364: Add an introduction page to modules
### Describe the proposed change(s).

We should add an instroduction page to the renovate modules section, so it's the first page instead of the datasources index.
Currently you need to close the datasources section to get to other modules.

//cc @HonkingGoose

### PR DESCRIPTION
**Pull request by viceice**
## Changes
Add a modules landing page, so we don't do to datasources when clicking on modules

## Context
- closes #28364

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

**Comment by viceice** [comment]
image

File: docs/usage/modules/index.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
+---
+title: Modules
```
Comment (HonkingGoose) [inline docs/usage/modules/index.md:2]:
```suggestion
title: Modules introduction
```

File: docs/usage/modules/index.md [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+---
+
+# Modules
```
Comment (HonkingGoose) [inline docs/usage/modules/index.md:5]:
```suggestion
# Modules introduction
```

File: docs/usage/modules/index.md [hunk 3]
```diff
@@ -0,0 +1,14 @@
+---
```
Comment (HonkingGoose) [inline docs/usage/modules/index.md:1]: Do we need this frontmatter?

```markdown
---
title: Modules
---
```

The markdown file has a H1 heading, which I guess/hope would be enough for Material for MkDocs to understand what title the page should have?

Read the [Material for MkDocs, reference, setting the page title](https://squidfunk.github.io/mkdocs-material/reference/#setting-the-page-title) to learn more.

Comment (viceice) [reply docs/usage/modules/index.md:1]: maybe you're right. will check it tomorrow

Comment (viceice) [reply docs/usage/modules/index.md:1]: if i remove it, then the pages plugin uses `Introduction` from `.pages` as page title, which isn't wanted here

File: docs/usage/modules/.pages [hunk 4]
```diff
@@ -1,3 +1,6 @@
 title: Renovate Modules
+nav:
+  - Introduction: index.md
+  - ...
```
Comment (secustor) [inline docs/usage/modules/.pages:4]: Is this on purpose?

Comment (viceice) [reply docs/usage/modules/.pages:4]: yes, so introduction is first page and then the rest is generated 😏

**Review by secustor — APPROVED** [review]

**Pull request merged**